src/lib.rs

```diff
@@ -60,5 +60,6 @@ pub use common::*;
 pub use decoder::{Decoder, Reader, OutputInfo, StreamingDecoder, Decoded, DecodingError, Limits};
 #[cfg(feature = "png-encoding")]
 pub use encoder::{Encoder, Writer, EncodingError};
+pub use filter::FilterType;
```

Comment:
Regarding this, all other members of `Info` are public members in `pub use common::*`. Not having a public re-export of `deflate::Compression` but using that as a public member in a struct feels a bit strange. Two possibilities:

1. Commit to `deflate::Compression` and add a public export.
2. Create a simple `struct` wrapper like `pub struct Compression(deflate::Compression);` to `common` which hides the representation, and add a `impl From<deflate::Compression> for Compression` as well as a replica of the constants. Replace the info attribute to use that type instead, and unpack its representation in the encoder code.

On the plus side 1. is simpler but one should consider that [`deflate::CompressionOptions`](https://docs.rs/deflate/0.7.19/deflate/struct.CompressionOptions.html) allow even more control over the output compression. It would be easier to extend option 2. in a backwards compatible way to introduce these additional possibilities at a future point. At the same time, method 2. would not require adding an extra crate to every users dependency list (in edition 2015) or crate usage (in edition 2018). Basically a question of whether the additional consistency of 2. comes at a cost of convenience.